pkg/importer/dataStream_test.go

```diff
@@ -79,239 +172,99 @@ func TestNewDataStream(t *testing.T) {
 }
 
 func Test_dataStream_Read(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		Qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
-
-	type args struct {
-		buf []byte
-	}
 	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		want    int
-		wantErr bool
+		name     string
+		testFile string
+		wantErr  bool
 	}{
-	// TODO: Add test cases.
+		{
+			name:     "successful read of datastream buffer",
+			testFile: "sample.tar.xz",
+			wantErr:  false,
+		},
+		{
+			name:     "failure on invalid read of empty datastream buffer",
+			testFile: "",
+			wantErr:  true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			d := dataStream{
-				Url:         tt.fields.Url,
-				Readers:     tt.fields.Readers,
-				buf:         tt.fields.buf,
-				qemu:        tt.fields.Qemu,
-				Size:        tt.fields.Size,
-				accessKeyId: tt.fields.accessKeyId,
-				secretKey:   tt.fields.secretKey,
-			}
-			got, err := d.Read(tt.args.buf)
+			ds, testBytes, expectedSize := createDataStreamBuf("", "", "", tt.testFile)
```

Comment:
You may run into trouble closing over `tt.testfile`.  Be sure that you get the expected value for each test iteration.  If you don't, it's likely because `tt.testFile` is being shared between tests.  The behavior appears at `tt.testFile` always having the last value in the slice, regardless of which element in the slice is being executed.

Usually safer to get an in-scope copy of the value:
```
for _, tt := range tests {
        tf := tt.testFile
	t.Run(tt.name, func(t *testing.T) {
                 ds, testBytes, expectedSize := createDataStreamBuf("", "", "", tf)
```